src/nunit.analyzers/DiagnosticSuppressors/DereferencePossiblyNullReferenceSuppressor.cs

```diff
@@ -61,38 +61,116 @@ private static bool ShouldBeSuppressed(SyntaxNode node)
+                BlockSyntax? block = statement.Parent as BlockSyntax;
 
                 // If the statement is inside an Assert.Multiple NUnit doesn't throw
                 // flow continues with actual null values and therefore these shouldn't be suppressed.
                 if (!AssertHelper.IsInsideAssertMultiple(statement))
                 {
                     if (IsKnownToBeNotNull(currentNode) ||
-                        (parent is not null && IsValidatedNotNullByPreviousStatementInSameBlock(possibleNullReference, parent, statement)))
+                        (block is not null && IsValidatedNotNullByPreviousStatementInSameBlock(possibleNullReference, block, statement)))
                     {
                         return true;
                     }
                 }
 
-                currentNode = parent;
+                currentNode = statement.Parent;
             }
 
             return false;
         }
 
-        private static bool IsValidatedNotNullByPreviousStatementInSameBlock(string possibleNullReference, BlockSyntax parent, StatementSyntax statement)
+        /// <summary>
+        /// Determine which side of the conditional expression if failing nullability.
+        /// </summary>
+        /// <remarks>
+        /// We only try to detect simple cases where the operands being tested for <see langword="null"/> is one of the operands being returned:
+        /// <code>variable is not null ? variable : otherExpression</code>
+        /// </para>
+        /// We recognize the 'is' pattern operations and normal equality.
+        /// </remarks>
+        /// <param name="conditionalExpression">Conditional expression to investigate.</param>
+        /// <returns>Either the 'WhenTrue' or the 'WhenFalse' part of the <paramref name="conditionalExpression"/>.</returns>
+        private static ExpressionSyntax? DetermineConditionalExpressionPath(ConditionalExpressionSyntax conditionalExpression)
+        {
+            string? testedAgainstNullExpression;
+            bool isNot = false;
+
+            if (conditionalExpression.Condition is IsPatternExpressionSyntax patternExpression &&
+                patternExpression.Expression is IdentifierNameSyntax or MemberAccessExpressionSyntax)
+            {
+                // identifier is
+                testedAgainstNullExpression = patternExpression.Expression.ToString();
+                PatternSyntax pattern = patternExpression.Pattern;
+                if (pattern is UnaryPatternSyntax unaryPattern && unaryPattern.IsKind(SyntaxKind.NotPattern))
+                {
+                    // 'expression' is not
+                    isNot = true;
+                    pattern = unaryPattern.Pattern;
+                }
+
+                if (pattern is not ConstantPatternSyntax constantPattern || !constantPattern.Expression.IsKind(SyntaxKind.NullLiteralExpression))
+                {
+                    return null;
+                }
+
+                // identifier is (not) null
```

Comment:
I think this is a bit odd place for the comment. Should this have been on top of line 128?

Reply:
I was indicating that if we get here, it was a 'is (not) null' expression and not an 'is some-other-constant'.
Updated comments.